src/lib/project.js

```diff
@@ -1,57 +1,170 @@
-  const lang = 'ts';
+async function project({ name, ui }) {
   const isWindows = process.platform === 'win32';
 
   if (fs.existsSync(name)) {
-    console.error(_red(`Directory already exists. Not proceeding`));
+    console.error(red(`Directory already exists. Not proceeding`));
     return;
   }
 
   // Git must be initialized before running `npm install` b/c Husky runs an
   // NPM `prepare` script to set up its pre-commit hook within `.git`.
   // Check before fetching project template, to not leave crud on user's system.
   if (!sh.which('git')) {
-    console.error(_red('Please ensure Git is installed, then try again.'));
+    console.error(red('Please ensure Git is installed, then try again.'));
     return;
   }
 
-  if (!(await fetchProjectTemplate(name, lang))) return;
+  let res;
+  if (!ui) {
+    try {
+      res = await prompt({
+        type: 'select',
+        name: 'ui',
+        choices: ['svelte', 'next', 'nuxt', 'empty', 'none'],
+        message: (state) => {
+          // Make the step text green upon success, else use the reset color.
+          const style =
+            state.submitted && !state.cancelled ? state.styles.success : reset;
+          return style('Create an accompanying UI project too?');
+        },
+        prefix: (state) => {
+          // Show a cyan question mark when not submitted.
+          // Show a green check mark if submitted.
+          // Show a red "x" if ctrl+C is pressed (default is a magenta).
+          if (!state.submitted) return state.symbols.question;
+          return !state.cancelled
+            ? state.symbols.check
+            : red(state.symbols.cross);
+        },
+      });
+    } catch (err) {
+      // If ctrl+c is pressed it will throw.
+      return;
+    }
 
-  // Create a keys dir because we excluded it from Git
-  fs.ensureDirSync(`${name}/keys`);
+    ui = res.ui;
+  }
+
+  sh.mkdir('-p', name); // Create path/to/dir with their desired name
+  sh.cd(name); // Set dir for shell commands. Doesn't change user's dir in their CLI.
 
-  // Set dir for shell commands. Doesn't change user's dir in their CLI.
-  sh.cd(name);
+  // If user wants a UI framework installed alongside their smart contract,
+  // we'll create this dir structure:
+  //   /<name>     (with .git)
+  //     ui/
+  //     contracts/
+  // - We use NPM for the UI projects for consistency with our smart contract
+  //   project, as opposed to Yarn or PNPM.
+  // - spawnSync with stdio:inherit allows the child process to be interactive.
+  if (ui) {
+    switch (ui) {
+      case 'svelte':
+        // `-y` installs the latest version of create-svelte without prompting.
+        spawnSync('npm', ['create', 'svelte@latest', '-y', 'ui'], {
+          stdio: 'inherit',
+          shell: true,
+        });
+        break;
+      case 'next':
+        // https://nextjs.org/docs/api-reference/create-next-app#options
+        spawnSync('npx', ['create-next-app@latest', 'ui', '--use-npm'], {
```

Comment:
Probably want to add  --ts here to use typescript

Reply:
We can prompt to ask if they want TypeScript, similar to how SvelteKit and NuxtJS do currently. 

(I've also considered a `--ts` flag on the `zk project` command, but we wouldn't be able to pass this to SvelteKit and NuxtJS installation scripts b/c they don't allow specifying this non-interactively. So adding a prompt for NextJS users is probably the best solution.)

Preference to land this so work can start on headers and imports, then to add this later.

Reply:
Created an issue to prompt NextJS users to ask if they want TypeScript: https://github.com/o1-labs/zkapp-cli/issues/278